### DISCUSSION TITLE
Static HTML export: how to make external linking to dynamic routes work (without getStaticProps)?

### DISCUSSION THREAD

**Discussion by FooBarWidget** [vercel/next.js#28249] [Help]
I'm making a single-page application that queries a Go backend server for data. I want to deploy the frontend as a bunch of static HTML files, without a Node.js server. Next.js's static HTML export feature — `next build && next export` — works for the most part. But I how do I make linking to dynamic routes work? Dynamic routes are not directly backed by an .html file, so any links to that dynamic route return 404 errors.

For example let's say I have `applications/[id].tsx`. Linking to `http://my-app.com/applications` works, because the web server serves /applications.html. Linking to `http://my-app.com/applications/123` doesn't work, because there is no /applications/123.html.

How do I make this work? Note that I cannot use getStaticProps to prerender: these pages query their data from a backend server written in Go.

One possibility that I can think of, is to make Next.js use hash routing (e.g. `http://my-app.com/#!/applications/123`). Is this possible, and if so, how do I make this work?

**Comment by miloofcroton** [comment]
I just posted about this here: https://github.com/vercel/next.js/discussions/28752

My best advice is to use the query params, but that is less than ideal because then you don't have pretty URLs.
